### ISSUE TITLE
Should Elsa persist FinishOutput or a JSON token object in the database? 

### ISSUE THREAD

**Issue by craigfowler** [elsa-workflows/elsa-core#650]
_This issue began life as a discussion in #552_.  Elsa currently reads & writes instances of `Elsa.Models.FinishOutput` directly from/to the persistence layer.  Aside from the bug reported above, this should be revisited to decide whether that's a good implementation or not.

## Options discussed
* Elsa could continue to persist the `Elsa.Models.FinishOutput` instance as a strongly-typed object
* Elsa could instead persist a `JToken` object and use mapping or serialisation where applicable

**Comment by sfmskywalker** [comment]
Unless I misunderstood the original MongoDB serialization issue, I would suggest to continue persisting `FinishOutput` as a strongly-typed object, including its value, making sure to include its .NET assembly-qualified type name. This would keep things consistent.

**Comment by craigfowler** [comment]
I did wonder.  I generated this based on [this comment](https://github.com/elsa-workflows/elsa-core/issues/552#issuecomment-754028524), in case it implied that "the future of persistence is intended to be JToken-based".

But indeed, I'd advocate for serializing strongly typed instances of classes rather than JToken anyway, so I'm totally happy for the answer to the issue to be _"No, they shouldn't"_ and for this to go straight to the land of WONTFIX.
